test/src/register_types.cpp

```diff
@@ -26,6 +26,9 @@ void initialize_example_module(ModuleInitializationLevel p_level) {
 	ClassDB::register_class<Example>();
 	ClassDB::register_class<ExampleVirtual>(true);
 	ClassDB::register_abstract_class<ExampleAbstract>();
+
+	GDREGISTER_VIRTUAL_CLASS(ExamplePureVirtualBase);
+	GDREGISTER_CLASS(ExamplePureVirtual);
```

Comment:
```suggestion
	ClassDB::register_abstract_class<ExamplePureVirtualBase>();
	ClassDB::register_class<ExamplePureVirtual>();
```
Use same syntax for all, also this is an abstract class, not a virtual one, methods are pure virtual, but classes are abstract

See if that fixes it in fact

Reply:
```suggestion
	ClassDB::register_abstract_class<ExampleAbstractBase>();
	ClassDB::register_class<ExampleConcrete>();
```
As per the above changes

Reply:
I've wanted to switch to using the macros for everything here, because that would be more consistent with Godot. But that's probably best to do in a dedicated PR, for this PR keeping this chunk of code consistent with itself makes sense (like @AThousandShips is suggesting).

Reply:
Could do it here too but this change is not just the consistency